docs/installation/viam-micro-server-dev.md

```diff
@@ -20,275 +20,262 @@ aliases:
+1. Generate a new module skeleton from [this template](https://github.com/viamrobotics/micro-rdk/tree/main/templates/module):
 
    ```sh { class="command-line" data-prompt="$"}
-   cargo install espup
-   cargo install cargo-espflash
-   cargo install ldproxy
+   cargo generate --git https://github.com/viamrobotics/micro-rdk.git
    ```
 
-6. Download and install the ESP-RS toolchain:
+   Select `templates/module` when prompted, give the module a name of your choice, and answer any additional prompts.
+
+1. Change directories into the generated tree:
 
    ```sh { class="command-line" data-prompt="$"}
-   espup install -s -f ~/esp/export-rs.sh -v 1.75.0
+   cd <your-path-to/your-module-directory>
    ```
 
-7. Activate the ESP Rust toolchain, run the following command to source (`.`) the activation script `export-rs.sh`:
+1. If you wish to use revision control for the module, this is the best time to initialize a git repository and commit all the generated files.
+   There are no secrets in a newly generated module repository:
 
    ```sh { class="command-line" data-prompt="$"}
-   . $HOME/esp/export-rs.sh
+   git add .
+   git commit -m "initial commit"
    ```
 
-   {{< alert title="Tip" color="tip" >}}
+1. Develop the module by defining `structs` which implement the necessary `traits` and adding tests and registration hooks for them, per the walkthrough.
 
-To avoid conflicts with other toolchains, adding this command to your .bashrc or .zshrc is not recommended.
-Instead, save this command to run in any future terminal session where you need to activate the ESP-IDF development framework.
-
-{{< /alert >}}
+1. To consume the module, follow the "Create a Project" workflow in a different directory, and register your module in the `[dependencies]` section of the project's `Cargo.toml` file, then build and flash the project.
+   The module will now be available for use by adding it to your machine configuration on the [Viam App](https://app.viam.com).
 
 {{% /tab %}}
-{{% /tabs %}}
-
-## Install `viam-micro-server`
-
-1. Create a new machine
-
-   Navigate to [the Viam app](https://app.viam.com) and [add a new machine](/cloud/machines/#add-a-new-machine) in your desired location.
+{{% tab name="viam-micro-server Development" %}}
 
-   Click on the name of the machine to go to the machine's page.
-
-1. [Start Docker](https://docs.docker.com/config/daemon/start/) on your development machine.
-   If you haven't already, complete Docker's [Linux post installation steps](https://docs.docker.com/engine/install/linux-postinstall/) to set up Docker to run whenever your system boots up.
-
-1. Generate a new project from [this template](https://github.com/viamrobotics/micro-rdk/tree/main/templates/project) to create a new `viam-micro-server` project to upload to your ESP32 by running:
+1. If you have not done so before, clone the `viamrobotics/micro-rdk` git repository (optionally specifying a path for the new clone) and `cd` into the clone:
```

Comment:
They'll know if they have done that ans skip the step as appropriate - I think fair to skip
```suggestion
1. Clone the `viamrobotics/micro-rdk` repository (optionally specifying a path for the new clone) and `cd` into the clone:
```